kubernetes/core/src/main/kotlin/com/google/kubernetes/tools/core/KubectlExecutorService.kt

```diff
@@ -0,0 +1,166 @@
+            process.waitFor(2, TimeUnit.SECONDS)
+            process.exitValue() == 0
+        } catch (e: Exception) {
+            false
+        }
+
+    /**
+     * Parse through command line arguments and try to execute the command
+     */
+    fun executeKubectl(settings: KubectlExecutorSettings): KubectlProcess {
+
+        val commandList = mutableListOf<String>()
+
+        with(commandList) {
+            add(kubectlExecutablePath)
+            add(settings.executionMode.modeFlag)
+            settings.executionFlags.forEach {
+                add(it)
+            }
+        }
+
+        try {
+            val processBuilder = ProcessBuilder()
+            val startedProcess = processBuilder.command(commandList).start()
+            val kubectlProcess = KubectlProcess(
+                    startedProcess,
+                    commandList.joinToString(" ")
+            )
+            // track event based on execution mode.
+            UsageTrackerService.getInstance().trackEvent(KUBECTL_SUCCESS).ping()
+            return kubectlProcess
+        } catch (e: Exception) {
+            val kubectlFailEventName = KUBECTL_FAIL
+
+            UsageTrackerService.getInstance().trackEvent(kubectlFailEventName)
+                    .addMetadata(
+                            METADATA_ERROR_MESSAGE_KEY, e.javaClass.name
+                    ).ping()
+            // re-throw exception to make sure a user sees run resulted in an error and sees error
+            // message.
+            throw e
+        }
+    }
+
+    /**
+     * Check if kubectl is available, Check if the flags supplied are valid. Create and start the
+     * command line process passed in
+     */
+
+    fun processOutputToString(executingProcess: Process): String {
+        val reader = BufferedReader(InputStreamReader(executingProcess.inputStream))
+        val builder = StringBuilder()
+        var line = reader.readLine()
+        while (line != null) {
+            builder.append(line)
+            line = reader.readLine()
+            //so that there's no trailing or leading whitespace
+            if (line != null) builder.append(System.getProperty("line.separator"))
+        }
+        return builder.toString()
+    }
+
+    /**
+     * Check if kubectl is available, Check if the flags supplied are valid. Create and start the
+     * command line process passed in
+     */
+    fun startProcess(
+        executionMode: KubectlExecutorSettings.ExecutionMode,
+        configurationFlags: List<String>
+    ): String {
+
+        if (!isKubectlAvailable()) {
+            throw ExecutionException(message("kubectl.not.on.system.error"))
+        }
+
+        val executorSettings = KubectlExecutorSettings(
+                executionMode = executionMode,
+                executionFlags = configurationFlags
+        )
+
+        val executingProcess = executeKubectl(executorSettings).process
+
+        executingProcess.waitFor(2, TimeUnit.SECONDS)
```

Comment:
We probably cannot know for sure the execution time, how about:
* use `waitFor()` method
* mention in comments this method waits until process completes and needs to be run in a separate backrground thread?